packages/jest-haste-map/src/index.js

```diff
@@ -820,8 +820,17 @@ class HasteMap extends EventEmitter {
     }
 
     if (this._whitelist) {
-      const match = filePath.match(this._whitelist);
-      return !match || match.length > 1;
+      const whitelist = this._whitelist;
+      const match = whitelist.exec(filePath);
+      const matchEndIndex = whitelist.lastIndex;
```

Comment:
[Docs says `lastIndex` stays zero](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/exec) if the regexp doesn't have the `g` flag. Isn't this a problem? Can we use the result `match` to figure it out instead?